up42/viztools.py

```diff
@@ -308,29 +322,42 @@ def map_results(
                 "You first need to download the results via job.download_results()!"
             )
 
-        # Add features to map.
-        # Some blocks store vector results in an additional geojson file.
-        # pylint: disable=not-an-iterable
-        json_fp = [fp for fp in self.results if fp.endswith(".geojson")]
-        if json_fp:
-            json_fp = json_fp[0]
-        else:
+        f_paths = []
+        if isinstance(self.results, list):
+            # Add features to map.
+            # Some blocks store vector results in an additional geojson file.
             # pylint: disable=not-an-iterable
-            json_fp = [fp for fp in self.results if fp.endswith(".json")][0]
+            json_fp = [fp for fp in self.results if fp.endswith(".geojson")]
+            if json_fp:
+                json_fp = json_fp[0]  # why only one element is selected?
+            else:
+                # pylint: disable=not-an-iterable
+                json_fp = [fp for fp in self.results if fp.endswith(".json")][0]
+            f_paths = self.results
+        elif isinstance(self.results, dict):
+            # pylint: disable=unsubscriptable-object
+            json_fp = self.results["merged_result"][0]
```

Comment:
Nice, so here we do not need anymore to get rid of merging results in JobCollection as @chrieke mentioned? right?

Reply:
I mean, we have access to the merged results, so better to leverage that!

Reply:
This is important not to create an extra ticket for removing that method then. Since it will then affect this one as well :D

Reply:
For the visualization we decided to iterate over the jobs and visualize each result independently, so the merged result is not required here. We might remove the merged result alltogether, but separate topic. I think here you need to iterate over the list of n job results instead and create n map objects.

Reply:
@chrieke  What would be the plus (and minuses) for now using `merged` result?
From current state of code and my perspective, having this file is proving to be rather beneficial. This enables us to create a single dataframe without looping and appending business, which then is used for viz.

Reply:
To clarify, merged is the merged result of all jobs, not of multiple jsons of one job.

If we merge the result and visualize only 1 map, this is at hand:
- We need to merge together also the geojson files, currently only data.json is merged together.
- Would blow up a single visualization very quickly, memory, code & usability wise (Long list of features, aois all over the globe etc.). See the ticket discussion. We decided to forego this question and do option, iterating over each job and visualizing it separatly, having multiple maps. 
- OSM block situation with multiple geojsons is new to me and not handled in the current code. Should have a separate discussion probably.